conans/client/cmake.py

```diff
@@ -115,18 +118,24 @@ def flags(self):
         op_system = str(self._settings.os) if self._settings.os else None
         arch = str(self._settings.arch) if self._settings.arch else None
         comp = str(self._settings.compiler) if self._settings.compiler else None
-        comp_version = self._settings.compiler.version
+        comp_version = self._settings.compiler.version     
+
+        try:
+            op_system_version = str(self._settings.os.version) if self._settings.os.version else None
```

Comment:
This is defined for some settings not existing in the default settings.yml, which is a bit confusing. Maybe it deserves some comment, with some examples of such os.version. Maybe even some testing. 
A classification of OS could be difficult, what would be the version of Ubuntu 14.04? 14.04?, How is it managed with other distros?

I know that some users are adding settings to differentiate for example between RedHat major distros (6, 7), but this is customized for their use case.